whats_fresh/whats_fresh_api/templates/product.html

```diff
@@ -97,15 +103,66 @@ <h4 id='error_tag' class='light warning'></h4>
 {% endblock content %}
 {% block footer %}
 <script>
+
     var preparation_options = {{ json_preparations|safe }};
 
     number_of_preparations = 0;
 
+    function popitup(url) {
+      $.ajax({
+        type: "GET",
+        url: url,
+        headers: { 'X-CSRFToken': '{{ csrf_token }}' },
+        success: function(data) {
+
+          $("#dialog-form").html(data).dialog({modal:true}, {title: "New Preparation"}, {resizable: true}, {width: 'auto'}, {draggable: false}).dialog('open');
+        }
+      });
+      return false;
+    }
+
+
+    $("#dialog-form").on('submit', function (e) {
+
+         e.preventDefault();
+
+        $.ajax({
+            type: "POST",
+            url: '/entry/products/new/preparations/new',
+            data: $("#popup-form").serialize(),
+            headers: { 'X-CSRFToken': '{{ csrf_token }}' },
+            success: function(data)
+            {
+                if (data.constructor === "test".constructor) {       // checks if the data is a string
+                    $("#dialog-form").html(data);
+                    $('html, body').animate({
+                        scrollTop: $("#preparation_error").offset().top - 100
+                    }, 500);
+                    return
+                }
+
+                $("#dialog-form").dialog('close');
+                var prepName = data.name;
+                var prepId = data.id;
+
+                $('#inline_preparation').append($('<div class="popup_preparation">')
```

Comment:
What happens if we give these the class preparation as well? Would make the code on line 164 (``if ($( ".preparation:visible" ).length === 0 && $( ".popup_preparation:visible" ).length === 0``) more idiomatic, and generally make sense I think.